packages/flutter_tools/lib/src/template.dart

```diff
@@ -361,7 +361,22 @@ class Template {
           context['androidIdentifier'] = _escapeKotlinKeywords(androidIdentifier);
         }
 
-        final String renderedContents = _templateRenderer.renderString(templateContents, context);
+        Map<String, Object?> localContext = context;
+
+        // Ensure inputs are escaped when necessary.
+        if (finalDestinationFile.path.endsWith('.yaml')) {
+          // Use a copy of the context,
+          // since the original is used in rendering other templates.
+          localContext = Map<String, Object?>.of(localContext);
+
+          final String? description = localContext['description'] as String?;
+
+          if (description != null && description.isNotEmpty) {
+            localContext['description'] = escapeYamlString(description);
+          }
+        }
```

Comment:
I'm a little worried that we're conditionally mutating the `localContext` variable, and this behavior may be obscured and forgotten in the future. Instead, can you pull the escaping logic into a helper function with a dartdoc, so it's more clear when and why we would be escaping, and with which copy?

```suggestion
        final Map<String, Object?> localContext = finalDestinationFile.path.endsWith('.yaml')
          ? _createEscapedContextCopy(context)
          : context;
```

And then the inner block moved into `Map<String, Object?> _createEscapedContextCopy(Map<String, Object?> oldContext)`